### PR TITLE
Unify recovery portal between Asgardeo and IS

### PR DESCRIPTION
**Pull request by DilshanSenarath**
### Related Issues
- https://github.com/wso2/product-is/issues/16560

### CODE REVIEW

File: java/apps/recovery-portal/src/main/webapp/includes/branding-preferences.jsp [hunk 1]
```diff
@@ -52,262 +49,237 @@
     String cookiePolicyURL = "/authenticationendpoint/cookie_policy.do";
     String selfSignUpOverrideURL = "";
     String passwordRecoveryOverrideURL = "";
-    String layout = "default";
-    String layoutFileRelativePath = "includes/layouts/default/body.ser";
-    String customLayoutFileRelativeBasePath = "";
+    String layout = "centered";
+    String layoutFileRelativePath = "includes/layouts/centered/body.ser";
```
Comment (DilshanSenarath) [inline java/apps/recovery-portal/src/main/webapp/includes/branding-preferences.jsp:53]: get the layout var

File: java/apps/recovery-portal/src/main/webapp/includes/language-switcher.jsp [hunk 2]
```diff
@@ -63,6 +63,8 @@
             let hostnameTokens = window.location.hostname.split('.');
```
Comment (DilshanSenarath) [inline java/apps/recovery-portal/src/main/webapp/includes/language-switcher.jsp:63]: change the implementation to get hostname from url object

File: java/apps/recovery-portal/src/main/webapp/includes/product-footer.jsp [hunk 3]
```diff
@@ -30,22 +32,103 @@
...
+                        } else {
+                    %>    
+                        &copy; <script>document.write(new Date().getFullYear());</script> WSO2 LLC.
+                    <%
+                        }
+                    %>
+                    <%
+                        if (!shouldRemoveDefaultBranding) {
+                    %>
+                    <div class="powered-by-logo-divider">|</div>Powered by <div class="powered-by-logo" onclick="window.open('<%= StringEscapeUtils.escapeHtml4(productURL) %>', '_self', 'noopener,noreferrer,resizable')">
```
Comment (DilshanSenarath) [inline java/apps/recovery-portal/src/main/webapp/includes/product-footer.jsp:89]: should be a asgardeo extension

File: java/apps/recovery-portal/src/main/webapp/password-recovery-notify.jsp [hunk 4]
```diff
@@ -79,99 +103,22 @@
...
     }
 %>
 
-<%-- Data for the layout from the page --%>
-<%
-    layoutData.put("containerSize", "medium");
-%>
-
 <!doctype html>
-<html lang="en-US">
```
Comment (DilshanSenarath) [inline java/apps/recovery-portal/src/main/webapp/password-recovery-notify.jsp:95]: add lang

File: java/apps/recovery-portal/src/main/webapp/password-reset-success.jsp [hunk 5]
```diff
@@ -0,0 +1,122 @@
+<%--
+~ Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com). All Rights Reserved.
```
Comment (DilshanSenarath) [inline java/apps/recovery-portal/src/main/webapp/password-reset-success.jsp:2]: license headers

File: java/apps/recovery-portal/src/main/webapp/password-reset.jsp [hunk 6]
```diff
@@ -274,11 +421,281 @@
...
+            }
+
+            /**
+             * Function to validate against consecutive character validator.
+             */
+            function checkConsecutiveMatch(_password, limit, pattern) {
+
+                var _consValid = true;
+                if (limit > 0 &&
+                _password.match(pattern) && _password.match(pattern).length > 0) {
```
Comment (JayaShakthi97) [inline java/apps/recovery-portal/src/main/webapp/password-reset.jsp:654]:
```suggestion
                    _password.match(pattern) && _password.match(pattern).length > 0) {
```

File: java/apps/recovery-portal/src/main/webapp/self-registration-complete.jsp [hunk 7]
```diff
@@ -95,62 +233,190 @@
...
+                <% } %>
             </layout:component>
             <layout:component componentName="MainSection" >
+                <div class="ui green segment mt-3 attached">
+                        <h3 class="ui header text-center slogan-message mt-4 mb-6" data-testid="self-register-complete-page-header">
+                            <% if (StringUtils.isNotBlank(confirm)) { %>
+                            <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "account.verified.successfully")%>
+                            <% } else if (accountLockOnCreationEnabled) { %>
+                            <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "you.are.almost.there")%>
+                            <% } else  {%>
```
Comment (JayaShakthi97) [inline java/apps/recovery-portal/src/main/webapp/self-registration-complete.jsp:294]:
```suggestion
                            <% } else {%>
```

File: java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_de_DE.properties [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+Please.note.all.consents.are.mandatory=Bitte achten Sie darauf, dass alle Einwilligungen obligatorisch sind
+All.asterisked.consents.are.mandatory=Alle mit einem Stern (*) gekennzeichneten Einwilligungen sind obligatorisch
+After.signin.we.use.a.cookie.in.browser=Wenn Sie sich anmelden, verwenden wir einen Cookie in Ihrem Browser, um Ihre Sitzung zu verfolgen. Sie können unsere
+Cookie.policy=Cookie -Richtlinie
+For.more.details=für weitere Informationen lesen
+I.confirm.that.read.and.understood=Ich habe die
+Privacy.policy=Datenschutz-Bestimmungen gelesen und verstanden
+toc=Nutzungsbedingungen
+product.tagline=Erstellen Sie in wenigen Minuten nahtlose Anmeldeerlebnisse für Ihre Anwendungen
+product.slogan=Asgardeo ist eine IDAAS, die Entwickler befähigt, einen sicheren und reibungslosen Zugang zu implementieren.
```
Comment (JayaShakthi97) [inline java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_de_DE.properties:99]: Check this Asgardeo reference. Check other languages as well.

File: java/apps/recovery-portal/src/main/webapp/includes/footer.jsp [hunk 9]
```diff
@@ -17,4 +17,107 @@
...
+     * Extracts the domain from the hostname.
+     * If parsing fails, undefined will be returned.
+     */
+    function extractDomainFromHost() {
+
+        var domain = undefined;
+
+        /**
+        * Extract the domain from the hostname.
+        * Ex: If dev.accounts.asgardeo.io is parsed, `asgardeo.io` will be set as the domain.
```
Comment (JayaShakthi97) [inline java/apps/recovery-portal/src/main/webapp/includes/footer.jsp:112]: Shall we use a generic url example here?

File: java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_de_DE.properties [hunk 10]
```diff
@@ -0,0 +1,337 @@
+#
+# Copyright (c) 2023, WSO2 LLC. (https://www.wso2.org) All Rights Reserved.
```
Comment (JayaShakthi97) [inline java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_de_DE.properties:2]:
```suggestion
# Copyright (c) 2023, WSO2 LLC. (https://www.wso2.org)
```

File: java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_es_ES.properties [hunk 11]
```diff
@@ -0,0 +1,337 @@
+#
+# Copyright (c) 2023, WSO2 LLC. (https://www.wso2.org) All Rights Reserved.
```
Comment (JayaShakthi97) [inline java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources_es_ES.properties:2]: Check other places as well.
```suggestion
# Copyright (c) 2023, WSO2 LLC. (https://www.wso2.org) 
```

**Review by JayaShakthi97 — APPROVED** [review]

File: java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties [hunk 12]
```diff
@@ -129,13 +128,112 @@ error=Error
...
-lang.switcher.french=Fran�ais - France
 select.language=Select Language
+username.length.should.be=The username length should be between
+username.with.symbols=The username should consist of alphanumeric characters (a-z, A-Z, 0-9) and must include at least one letter.
+
+# Password Management
+forgot.password=Forgot Password?
+password.reset.with.username=Don't worry, it happens. We will send you an email to reset your password.
+password.reset.with.email=Don't worry, it happens. Please enter your email address. We will send you an email to reset your password.
+password.requirements=Your password should be more than 8 characters, have atleast one uppercase and lowercase letter and atleast one number.
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties:185]: There was an effort to refactor these error messages to keep those professional. Could you please check and update accordingly?

cc: @Achintha444

File: java/apps/recovery-portal/src/main/webapp/challenge-question-view.jsp [hunk 13]
```diff
@@ -148,15 +146,13 @@
...
-                productFooterFilePath = customLayoutFileRelativeBasePath + "/product-footer.jsp";
-            }
-            if (!new File(getServletContext().getRealPath(productFooterFilePath)).exists()) {
-                productFooterFilePath = "includes/product-footer.jsp";
-            }
+                File productFooterFile = new File(getServletContext().getRealPath("extensions/product-footer.jsp"));
+                if (productFooterFile.exists()) {
             %>
-            <jsp:include page="<%= productFooterFilePath %>" />
+            <jsp:include page="extensions/product-footer.jsp"/>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/challenge-question-view.jsp:152]:
```suggestion
                <jsp:include page="extensions/product-footer.jsp"/>
```

File: java/apps/recovery-portal/src/main/webapp/challenge-question-view.jsp [hunk 14]
```diff
@@ -148,15 +146,13 @@
...
+            <% } else { %>
+            <jsp:include page="includes/product-footer.jsp"/>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/challenge-question-view.jsp:154]:
```suggestion
                <jsp:include page="includes/product-footer.jsp"/>
```

File: java/apps/recovery-portal/src/main/webapp/error.jsp [hunk 15]
```diff
@@ -108,55 +109,62 @@
...
+                    <%
+                        if (config.getServletContext().getResource("extensions/error-tracking-reference.jsp") != null) {
+                    %>
+                            <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "with.tracking.reference.below")%>
+                        </p>
+                        <div class="ui divider hidden"></div>
+                        <jsp:include page="extensions/error-tracking-reference.jsp"/>
+                    <%
+                        }
+                    %>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/error.jsp:154]:
```suggestion
                <p class="text-left mt-0">
                    <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "need.help.contact.us")%>
                    <a href="mailto:<%= StringEscapeUtils.escapeHtml4(supportEmail) %>" target="_blank">
                        <span class="orange-text-color button"><%= StringEscapeUtils.escapeHtml4(supportEmail) %>
                        </span>
                    </a>
                    <%
                        if (config.getServletContext().getResource("extensions/error-tracking-reference.jsp") != null) {
                    %>
                            <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "with.tracking.reference.below")%>
                 </p>
                            <div class="ui divider hidden"></div>
                           <jsp:include page="extensions/error-tracking-reference.jsp"/>
                    <% } %>
```

File: java/apps/recovery-portal/src/main/webapp/errors/error.jsp [hunk 16]
```diff
@@ -30,13 +32,18 @@
 <%-- Branding Preferences --%>
 <jsp:directive.include file="../includes/branding-preferences.jsp"/>
 
+<%
+    String stat = IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "error.occurred");
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/errors/error.jsp:36]: Let's use more descriptive variable names.

```suggestion
    String status = IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "error.occurred");
```

File: java/apps/recovery-portal/src/main/webapp/errors/error_400.jsp [hunk 17]
```diff
@@ -30,9 +32,14 @@
 <%-- Branding Preferences --%>
 <jsp:directive.include file="../includes/branding-preferences.jsp"/>
 
+<%
+    String stat = IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "error.400");
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/errors/error_400.jsp:36]: Let's use more descriptive variable names. check the other changed places as well.

File: java/apps/recovery-portal/src/main/webapp/errors/error_401.jsp [hunk 18]
```diff
@@ -30,13 +32,18 @@
...
 <%-- Data for the layout from the page --%>
 <%
-    layoutData.put("containerSize", "medium");
+    layoutData.put("isResponsePage", true);
+    layoutData.put("isErrorResponse", true);
 %>
 
 <!doctype html>
-<html lang="en-US">
+<html>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/errors/error_401.jsp:46]: Let's add the lang attribute back, as it's required for i18n.

File: java/apps/recovery-portal/src/main/webapp/includes/footer.jsp [hunk 19]
```diff
@@ -17,4 +17,107 @@
 --%>
 
 <script src="libs/jquery_3.6.0/jquery-3.6.0.min.js"></script>
-<script src="libs/themes/default/semantic.min.js"></script>
+<script src="libs/themes/asgardio/semantic.min.js"></script>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/includes/footer.jsp:20]: Hope the theme name is updated to wso2is.

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-notify.jsp [hunk 20]
```diff
@@ -60,38 +63,63 @@
...
+                </p>
+                <p class="ui portal-tagline-description" data-testid="password-recovery-support-message">
+                    <%=IdentityManagementEndpointUtil.i18n(recoveryResourceBundle, "for.further.assistance.write.to")%>
+                    <a href="mailto:<%= StringEscapeUtils.escapeHtml4(supportEmail) %>"
+                    data-testid="password-recovery-resend-support-email"
+                    target="_blank">
+                    <span class="orange-text-color button">
+                        <%= StringEscapeUtils.escapeHtml4(supportEmail) %>
+                    </span>
+                    </a>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-notify.jsp:105]:
```suggestion
                    <a
                        href="mailto:<%= StringEscapeUtils.escapeHtml4(supportEmail) %>"
                        data-testid="password-recovery-resend-support-email"
                        target="_blank"
                    >
                        <span class="orange-text-color button">
                            <%= StringEscapeUtils.escapeHtml4(supportEmail) %>
                        </span>
                    </a>
```

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp [hunk 21]
```diff
@@ -197,15 +197,13 @@
...
-                productTitleFilePath = customLayoutFileRelativeBasePath + "/product-title.jsp";
-            }
-            if (!new File(getServletContext().getRealPath(productTitleFilePath)).exists()) {
-                productTitleFilePath = "includes/product-title.jsp";
-            }
+                File productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
+                if (productTitleFile.exists()) {
             %>
-            <jsp:include page="<%= productTitleFilePath %>" />
+            <jsp:include page="extensions/product-title.jsp"/>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp:203]:
```suggestion
                <jsp:include page="extensions/product-title.jsp"/>
```

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp [hunk 22]
```diff
@@ -197,15 +197,13 @@
...
+            <% } else { %>
+            <jsp:include page="includes/product-title.jsp"/>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp:205]:
```suggestion
                <jsp:include page="includes/product-title.jsp"/>
```

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp [hunk 23]
```diff
@@ -281,15 +279,13 @@
...
-                productFooterFilePath = "includes/product-footer.jsp";
-            }
+                File productFooterFile = new File(getServletContext().getRealPath("extensions/product-footer.jsp"));
+                if (productFooterFile.exists()) {
             %>
-            <jsp:include page="<%= productFooterFilePath %>" />
+            <jsp:include page="extensions/product-footer.jsp"/>
+            <% } else { %>
+            <jsp:include page="includes/product-footer.jsp"/>
+            <% } %>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp:288]:
```suggestion
                <jsp:include page="extensions/product-footer.jsp"/>
            <% } else { %>
                <jsp:include page="includes/product-footer.jsp"/>
            <% } %>
```

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp [hunk 24]
```diff
@@ -166,15 +166,13 @@
...
-                productTitleFilePath = "includes/product-title.jsp";
-            }
+                File productTitleFile = new File(getServletContext().getRealPath("extensions/product-title.jsp"));
+                if (productTitleFile.exists()) {
             %>
-            <jsp:include page="<%= productTitleFilePath %>" />
+            <jsp:include page="extensions/product-title.jsp"/>
+            <% } else { %>
+            <jsp:include page="includes/product-title.jsp"/>
+            <% } %>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp:175]:
```suggestion
                <jsp:include page="extensions/product-title.jsp"/>
            <% } else { %>
                <jsp:include page="includes/product-title.jsp"/>
            <% } %>
```

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp [hunk 25]
```diff
@@ -318,15 +316,13 @@
...
-                productFooterFilePath = "includes/product-footer.jsp";
-            }
+                File productFooterFile = new File(getServletContext().getRealPath("extensions/product-footer.jsp"));
+                if (productFooterFile.exists()) {
             %>
-            <jsp:include page="<%= productFooterFilePath %>" />
+            <jsp:include page="extensions/product-footer.jsp"/>
+            <% } else { %>
+            <jsp:include page="includes/product-footer.jsp"/>
+            <% } %>
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp:325]:
```suggestion
                <jsp:include page="extensions/product-footer.jsp"/>
            <% } else { %>
                <jsp:include page="includes/product-footer.jsp"/>
            <% } %>
```

File: java/apps/recovery-portal/src/main/webapp/password-recovery.jsp [hunk 26]
```diff
@@ -47,8 +50,11 @@
 <%
     boolean error = IdentityManagementEndpointUtil.getBooleanValue(request.getAttribute("error"));
     String errorMsg = IdentityManagementEndpointUtil.getStringValue(request.getAttribute("errorMsg"));
-    String username = request.getParameter("username");
+    String username = StringUtils.isNotEmpty(request.getParameter("username"))
+        ? Encode.forHtmlAttribute(request.getParameter("username"))
+        : "";
     boolean isSaaSApp = Boolean.parseBoolean(request.getParameter("isSaaSApp"));
+    String sp = request.getParameter("sp");
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/password-recovery.jsp:57]: Please check whether the parameter usage is encoded properly

File: java/apps/recovery-portal/src/main/webapp/password-reset-success.jsp [hunk 27]
```diff
@@ -0,0 +1,131 @@
...
+<jsp:directive.include file="includes/localize.jsp"/>
+
+<%-- Include tenant context --%>
+<jsp:directive.include file="tenant-resolve.jsp"/>
+
+<%-- Branding Preferences --%>
+<jsp:directive.include file="includes/branding-preferences.jsp"/>
+
+<%
+    String accessUrl = (String) request.getAttribute("accessUrl");
```
Comment (pavinduLakshan) [inline java/apps/recovery-portal/src/main/webapp/password-reset-success.jsp:40]: Do we really need string casting here?

File: java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties [hunk 28]
```diff
@@ -58,9 +58,8 @@ Skip=Skip
 Password.recovery.information.sent.to.the.email.registered.with.account=Password recovery information has been sent to the email registered with the account
 Password.cannot.be.empty=Password cannot be empty.
 Updated.the.password.successfully=Updated the password successfully
-Server.failed.to.respond=Server failed to respond.
+Server.failed.to.respond=Opps! Something went wrong.
```
[the remark this answers is not in the document]
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties:61]: Already fixed https://github.com/wso2/identity-apps/blob/1e9a07c9708cd0a0e7769467d7d9966163c83092/identity-apps-core/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties#L109

File: java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties (hunk 12 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/resources/org/wso2/carbon/identity/mgt/recovery/endpoint/i18n/Resources.properties:185]: As discussed with @Achintha444 , Now customers have the capability to modify this content from the branding text customization section.

File: java/apps/recovery-portal/src/main/webapp/challenge-question-view.jsp (hunk 13 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/challenge-question-view.jsp:152]: https://github.com/wso2/identity-apps/pull/6249/commits/080c0acd02ec45c380b656cb2cb276720bde0d3a

File: java/apps/recovery-portal/src/main/webapp/challenge-question-view.jsp (hunk 14 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/challenge-question-view.jsp:154]: https://github.com/wso2/identity-apps/pull/6249/commits/080c0acd02ec45c380b656cb2cb276720bde0d3a

File: java/apps/recovery-portal/src/main/webapp/error.jsp (hunk 15 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/error.jsp:154]: https://github.com/wso2/identity-apps/pull/6249/commits/b7d34cb9bfc135bc50de9d6bc70297b197b11c7c

File: java/apps/recovery-portal/src/main/webapp/errors/error.jsp (hunk 16 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/errors/error.jsp:36]: https://github.com/wso2/identity-apps/pull/6249/commits/fd41a2f51c572ffdede1185953f75a9c551a3730

File: java/apps/recovery-portal/src/main/webapp/errors/error_400.jsp (hunk 17 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/errors/error_400.jsp:36]: https://github.com/wso2/identity-apps/pull/6249/commits/015847bdf35b5e241b11121d62713211ad0950db

File: java/apps/recovery-portal/src/main/webapp/errors/error_401.jsp (hunk 18 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/errors/error_401.jsp:46]: https://github.com/wso2/identity-apps/pull/6249/commits/dd9575ae867c6b5e9f648b34523c0db6f01875b6

File: java/apps/recovery-portal/src/main/webapp/includes/footer.jsp (hunk 19 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/includes/footer.jsp:20]: Already changed https://github.com/wso2/identity-apps/blob/86d6527d2e33193ca09d0cc086c28a4d10eb4528/identity-apps-core/apps/recovery-portal/src/main/webapp/includes/footer.jsp#L20

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-notify.jsp (hunk 20 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-notify.jsp:105]: https://github.com/wso2/identity-apps/pull/6249/commits/6e9dc7f9df4d071da0d9bb4e3e6860bad476c67c

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp (hunk 21 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp:203]: https://github.com/wso2/identity-apps/pull/6249/commits/87aa75a31cd8430a3acd326f8fea7b5db758ef33

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp (hunk 22 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims-options.jsp:205]: https://github.com/wso2/identity-apps/pull/6249/commits/87aa75a31cd8430a3acd326f8fea7b5db758ef33

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp (hunk 24 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp:175]: https://github.com/wso2/identity-apps/pull/6249/commits/87aa75a31cd8430a3acd326f8fea7b5db758ef33

File: java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp (hunk 25 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/password-recovery-with-claims.jsp:325]: https://github.com/wso2/identity-apps/pull/6249/commits/87aa75a31cd8430a3acd326f8fea7b5db758ef33

File: java/apps/recovery-portal/src/main/webapp/password-recovery.jsp (hunk 26 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/password-recovery.jsp:57]: Already done https://github.com/wso2/identity-apps/blob/86d6527d2e33193ca09d0cc086c28a4d10eb4528/identity-apps-core/apps/recovery-portal/src/main/webapp/password-recovery.jsp#L61

File: java/apps/recovery-portal/src/main/webapp/password-reset-success.jsp (hunk 27 above)
Comment (DilshanSenarath) [reply java/apps/recovery-portal/src/main/webapp/password-reset-success.jsp:40]: request.getAttribute returns an object of type 'Object'. Hence, it is better to keep this casting. https://docs.oracle.com/javaee%2F6%2Fapi%2F%2F/javax/servlet/ServletRequest.html#getAttribute(java.lang.String)

**Pull request merged**